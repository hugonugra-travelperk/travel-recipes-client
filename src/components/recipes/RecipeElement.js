import { useNavigate } from "react-router-dom";
import { CardWrapper, CardHeader, CardBody, CardSubHeader} from "../base/Card";
import IngredientsList from "../ingredients/IngredientsList"
import {Button} from '../../components/base/Button'
import DeleteButton from "./DeleteButton";

export default function RecipeElement(props){
  let navigate = useNavigate();

  const editParams=()=>{
    navigate('/edit/' + props.recipe.id, {state: {recipe: props.recipe}})
  }

  const viewParams=()=>{
    navigate('/' + props.recipe.id, {state: {recipe: props.recipe}})
  }

  function DetailButton(){
    if (!props.detail){
      return(
        <Button onClick={()=>{ viewParams() }}>View</Button>
      )
    }
  }

  return(
    <CardWrapper>
      <CardHeader>
        {props.recipe.name}
      </CardHeader>
      <CardBody>
        {props.recipe.description}
        <CardSubHeader>
          Ingredients
        </CardSubHeader>
        <IngredientsList recipe_id={props.recipe.id} ingredients={props.recipe.ingredients}/>
        {DetailButton()}
        <Button onClick={()=>{ editParams() }}>Edit</Button>
        <DeleteButton handleUpdate={props.handleUpdate} recipe = {props.recipe}/>
      </CardBody>
    </CardWrapper>
  )
}
