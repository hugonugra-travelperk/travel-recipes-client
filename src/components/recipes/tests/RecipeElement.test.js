import {render, cleanup} from "@testing-library/react"
import "@testing-library/jest-dom";
import { BrowserRouter } from "react-router-dom";
import RecipeElement from "../RecipeElement"

it("renders", () => {
  const {asFragment} = render(
    <BrowserRouter>
      <RecipeElement
        handleUpdate={()=>{}} recipe = {{'id':'1', 'name': 'test text recipe',
          'ingredients': [{'id': '1', 'name': 'test text ingredient'}]}
        }
      />
    </BrowserRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
