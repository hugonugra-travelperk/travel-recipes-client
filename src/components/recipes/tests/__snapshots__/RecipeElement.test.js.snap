// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div
    class="sc-bdvvtL gaBRgU"
  >
    <header
      class="sc-gsDKAQ jGoJmq"
    >
      test text recipe
    </header>
    <div
      class="sc-dkPtRN PHDyg"
    >
      <header
        class="sc-hKwDye klnJXo"
      >
        Ingredients
      </header>
      <div
        class="sc-eCImPb gayBGf"
      >
        <ul>
          <li
            class="sc-jRQBWg kEHRpN"
          >
            <span
              class="sc-gKclnd CJqjG"
            >
              test text ingredient
            </span>
          </li>
        </ul>
      </div>
      <button
        class="sc-iCfMLu fQrYjb"
      >
        View
      </button>
      <button
        class="sc-iCfMLu fQrYjb"
      >
        Edit
      </button>
      <button
        class="sc-iCfMLu gMEgxH"
      >
         Delete 
      </button>
    </div>
  </div>
</DocumentFragment>
`;
